### PR TITLE
New `PomerolED` method to compute the 3-point fermion-boson susceptibilities

### PR DESCRIPTION
**Pull request by krivenko**
This pull request adds a new method `PomerolED.chi3_inu_inup()` that can be called as follows.

```python
ed = PomerolED(index_converter)
ed.diagonalize(H)

params = {}
# Block structure of the susceptibility to be computed
params['gf_struct'] = [('up', 2), ('dn', 2)]
# Inverse temperature
params['beta'] = 10.0
# Compute all \chi^3 blocks except for ("dn", "dn")
# When this parameter is omitted, all blocks are computed
params['blocks'] = set([("up", "up"), ("up", "dn"), ("dn", "up")])
# Number of *positive* fermionic Matsubara frequencies
params['n_inu'] = 50
# \chi^3 channel, one of 'PP', 'PH' and 'xPH'
params['channel'] = 'PH'
# Order of block indices in the definition of \chi^3, one of 'AABB' and 'ABBA'
params['block_order'] = 'AABB'

chi3 = ed.chi3_inu_inup(**params)
```

@dominikkiese, @Wentzell Can you suggest any changes and/or additions?

### CODE REVIEW

**Comment by dominikkiese** [comment]
Renamings currently break tests, reference files will have to be replaced.

**Review by dominikkiese — APPROVED** [review]
Looks good to me.

**Comment by krivenko** [comment]
@Wentzell Could you have a look too, please?

File: c++/chi3.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    bool compute_all_blocks = chi3_blocks.empty();
+
+    std::vector<std::vector<gf<Mesh, tensor_valued<4>>>> gf_vecvec;
+    std::vector<std::string> block_names;
+
+    static const std::array<Pomerol::Channel, 3> pom_channels = {Pomerol::PP, Pomerol::PH, Pomerol::xPH};
+
+    for (auto const &bl1 : gf_struct) {
+      auto &A    = bl1.first;
+      int A_size = bl1.second;
```
Comment (Wentzell) [inline c++/chi3.cpp]:
```suggestion
    for (auto const &[A, A_size] : gf_struct) {
```

File: c++/chi3.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      int s1     = A_size;
```
Comment (Wentzell) [inline c++/chi3.cpp]: Why two names?

File: c++/chi3.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      block_names.push_back(A);
+
+      std::vector<gf<Mesh, tensor_valued<4>>> gf_vec;
+      for (auto const &bl2 : gf_struct) {
+        auto &B    = bl2.first;
+        int B_size = bl2.second;
```
Comment (Wentzell) [inline c++/chi3.cpp]:
```suggestion
      for (auto const &[B, B_size] : gf_struct) {
```

File: c++/chi3.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  if (!matrix_h) TRIQS_RUNTIME_ERROR << "chi3_iw_inu: No Hamiltonian has been diagonalized";
+  compute_rho(p.beta);
+  compute_field_operators(p.gf_struct);
+
+  if (verbose && !comm.rank()) std::cout << "chi3_iw_inu: Filling output container" << std::endl;
+
+  auto filler = [&p, this](gf_view<w_nu_t, scalar_valued> chi3_el, auto const &pom_chi3) {
+    long mesh_index = 0;
+    for (auto w_nu : chi3_el.mesh()) {
```
Comment (Wentzell) [inline c++/chi3.cpp]:
```suggestion
    for (auto [w, nu] : chi3_el.mesh()) {
```

File: c++/chi3.cpp [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      if ((mesh_index++) % comm.size() != comm.rank()) continue;
+
+      auto w = std::complex<double>(std::get<0>(w_nu));
+      auto nu = std::complex<double>(std::get<1>(w_nu));
```
Comment (Wentzell) [inline c++/chi3.cpp]:
```suggestion
```

**Review by Wentzell — COMMENTED** [review]
I suggested a few cosmetic changes, please incorporate them as you see fit. Otherwise everything looks good!

**Pull request merged**